### PR TITLE
Refactor same-user checking logic into presenter

### PR DESCRIPTION
**Pull request by Arepo**
After discussion with Sam I've made the minimal change here, but would like to refactor the Presenter class to expect a View object and call methods on that, rather than either including or passing in bits of view logic.

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by tansaku** [comment]
looks great @Arepo - awesome smallness!

File: app/views/users/profile/_summary.html.erb [hunk 1]
```diff
@@ -67,7 +67,7 @@
                          type: 'button' %></p>
         </li>
     <% end %>
-    <% if current_user && current_user == presenter.user %>
+    <% if current_user && presenter.user_same_as?(current_user) %>
```
Comment (tansaku) [inline app/views/users/profile/_summary.html.erb]: could we even pull the nil check in there into the user_same_as?

File: app/views/users/profile/_summary.html.erb [hunk 2]
```diff
@@ -39,7 +39,7 @@
         </li>
     <% end %>
 
-    <% if presenter.display_hire_me && current_user != presenter.user %>
+    <% if presenter.display_hire_me && !presenter.user_same_as?(current_user) %>
```
Comment (tansaku) [inline app/views/users/profile/_summary.html.erb]: could the  `!presenter.user_same_as?(current_user)` be pulled into the presenter.display_hire_me method?  no need to expose the logic in the view right?  I guess maybe that could be another ticket and PR but then again ... up to you :-)

**Comment by diraulo** [comment]
@tansaku @Arepo Should I assume all other refactorings will be dealt with as part of a separate PR and merge this?

**Comment by Arepo** [comment]
I'll fix the first one and see if I can do the second as quickly, else I'll pass it to a new PR

**Comment by tansaku** [comment]
@Arepo great work - I think this is ready to go!

**Comment by tansaku** [comment]
@diraulo I think this can be merged - @Arepo did all the refactorings

**Comment by diraulo** [comment]
Thanks for this @Arepo :wink:

**Pull request merged**